ci-operator/step-registry/openstack/provision/etcd-on-local-disk/openstack-provision-etcd-on-local-disk-commands.sh

```diff
@@ -117,118 +144,209 @@ metadata:
         enabled: true
-        name: restorecon-var-lib-etcd.service
-EOF
-
-wait_for_machineconfig_done
-
-info 'INFO: Replacing the 98-var-lib-etcd MachineConfig with a modified version that removes the logic for creating and syncing the device and that prevents the nodes from rebooting...'
-oc replace -f - <<EOF
-apiVersion: machineconfiguration.openshift.io/v1
-kind: MachineConfig
-metadata:
-  labels:
-    machineconfiguration.openshift.io/role: master
-  name: 98-var-lib-etcd
-spec:
-  config:
-    ignition:
-      version: 3.2.0
-    systemd:
-      units:
+        name: migrate-to-local-etcd.service
       - contents: |
           [Unit]
-          Description=Mount /dev/vdb to /var/lib/etcd
-          Before=local-fs.target
-          Requires=systemd-mkfs@dev-vdb.service
-          After=systemd-mkfs@dev-vdb.service var.mount
-
-          [Mount]
-          What=/dev/vdb
-          Where=/var/lib/etcd
-          Type=xfs
-          Options=defaults,prjquota
-
+          Description=Relabel /var/lib/etcd
+ 
+          # Run after we've migrated any existing content, but before crio so
+          # etcd isn't running yet.
+          After=migrate-to-local-etcd.service
+          Before=crio.service
+ 
+          # Only if /var/lib/etcd is mounted
+          Requisite=var-lib-etcd.mount
+ 
+          [Service]
+          Type=oneshot
+          RemainAfterExit=yes
+ 
+          # Do a quick check of the mountpoint directory before doing a full recursive relabel
+          # If restorecon /var/lib/etcd would not relabel the directory, don't
+          # run the recursive relabel
+          ExecCondition=/bin/bash -c "[ -n \"$(restorecon -nv /var/lib/etcd)\" ]"
+ 
+          ExecStart=/usr/sbin/restorecon -R /var/lib/etcd
+ 
           [Install]
-          WantedBy=local-fs.target
+          RequiredBy=var-lib-etcd.mount
         enabled: true
-        name: var-lib-etcd.mount
+        name: relabel-var-lib-etcd.service
 EOF
+info "INFO: Applying 98-var-lib-etcd MachineConfig"
+oc create -f /tmp/98-var-lib-etcd.yaml
+
+# Verify:
+# 
+# - The cluster is operating correctly
+# - /var/lib/etcd is a new mount
+# - /var/lib/etcd has the correct SELinux labels
+wait_for_machineconfig_done
+check_etcd_mounted_on_local_disk
+info "INFO: Moving etcd on local disk is complete and the cluster is healthy!"
+
+if [[ "${ETCD_ON_LOCAL_DISK_FULL_TEST}" == "false" ]]; then
+    info "INFO: Test is complete, exiting..."
+    exit 0
+fi
+
+info "This tests that the MachineConfig operates correctly when configured on a machine with no ephemeral disk."
+# Preparation:
+#
+# - Update CPMS to remove local etcd
+# - Wait for rollout to complete
+info "INFO: Remove additionalBlockDevices to the control plane machineset"
+oc patch ControlPlaneMachineSet/cluster -n openshift-machine-api --type json -p '[{"op": "remove", "path": "/spec/template/machines_v1beta1_machine_openshift_io/spec/providerSpec/value/additionalBlockDevices"}]'
```

Comment:
As we discovered: we need to change the flavor here to one without an ephemeral disk or we'll get one anyway.